src/CodeGen_Hexagon.cpp

```diff
@@ -1445,27 +1447,25 @@ void CodeGen_Hexagon::visit(const Call *op) {
         builder->CreateCondBr(any_true, true_bb, false_bb);
 
         builder->SetInsertPoint(true_bb);
-        codegen(load_expr);
+        Value *true_value = codegen(load_expr);
         builder->CreateBr(after_bb);
+        // The true value might have jumped to a new block (e.g. if there was a
```

Comment:
This comment looks like it comes from the Select visitor, update? Or just remove it, it's probably not necessary.